### ISSUE TITLE
Question about use softmax and Linear in minibatch of PyG

### ISSUE THREAD

**Issue by jiaruHithub** [pyg-team/pytorch_geometric#1508]
## ❓ Questions & Help

Hi,Friends.
here,I have a ```feature_matrix``` of a graph.I concate the output of three GCN layers.If I wan to use softmax and Linear,I should usr code as follow:
```
x = torch.stack([x1,x2,x3],dim=1)
x_in = x.view(x.size(0), 3, -1)
x_sorce_base = torch.nn.Linear(x_in)
attention_score = torch.nn.functional.softmax(x_sorce_base, dim=0)
x_out = x_in*attention_score 
```
PyG has ADVANCED MINI-BATCHING.The feature_matrix always contains many graph(size is ```N_graph*n_nodes,dimension```).
How do I do the same operation on a matrix that contains multiple graphs.(Whether batch is available for distinguishing?)
I hope to do this same operation separately for each matrix in a matrix that contains multiple graphs.
Thank you!

**Comment by rusty1s** [comment]
If you want to apply `softmax` in the first dimension of a PyG batch, you want to use `torch_geometric.utils.softmax`.

**Comment by jiaruHithub** [comment]
> If you want to apply `softmax` in the first dimension of a PyG batch, you want to use `torch_geometric.utils.softmax`.

Fine thank you.
But how can I assign ```dim```?

**Comment by rusty1s** [comment]
`torch_geometric.utils.softmax` is only useful when operating in the node dimension. If you want to apply a softmax in the feature dimension, you can use `torch.nn.functional.softmax`.
